### DISCUSSION TITLE
Support for custom interface fields on relationships.

### DISCUSSION THREAD

**Discussion by eliashussary** [directus/directus#3985] [Archive]
I've tried to implement a custom interface on a relationship field and it doesn't seem to work. 

I was hoping to achieve two things with this;

1. Paginate related items in detail view for o2m collections
   - Currently all records are retrieved, this is not quite ideal for records where there may be alot of records to display.
2. The inverse of that, for o2o or m2o, i'd like the ability to see the whole record, or a larger portion of the record
   - The default is currently a simple templated display `{{fieldA}} {{fieldB}}` etc.

**Comment by rijkvanzanten** [comment]
> [...] and it doesn't seem to work.

What was the problem? Custom interfaces should work as expected, regardless of interface type. 

> Paginate related items in detail view for o2m collections
> Currently all records are retrieved, this is not quite ideal for records where there may be alot of records to display.

This is something the "regular" o2m interface should do as well! So instead of a custom interface, that's more than welcome as an update to the core o2m interface 🙂 

> The inverse of that, for o2o or m2o, i'd like the ability to see the whole record, or a larger portion of the record
> The default is currently a simple templated display {{fieldA}} {{fieldB}} etc.

Am I right in understanding that's the same as https://github.com/directus/directus/discussions/3474?

**Reply by eliashussary** [reply]
> Am I right in understanding that's the same as #3474?

#3474 sounds pretty much the same, but in my case a read-only state, which i assume it would do aswell!

> What was the problem? Custom interfaces should work as expected, regardless of interface type.

Not entirely sure. I haven't had a chance to dig deep into `/app` yet - still trying to understand the architecture of it.

I basically copied the regular interface's config, but it won't show up. I can get it to work with standard fields though!

```ts
import InterfaceComponent from "./interface.vue";

export default {
  id: "custom",
  name: "Custom",
  description: "This is my custom interface!",
  icon: "box",
  component: InterfaceComponent,
  relationship: "m2o",
  types: ["uuid", "string", "text", "integer", "bigInteger"]
};
```
^ Adapted from: https://github.com/directus/directus/blob/c258ba6b9e1a330c1a954e88cdfd50dec2759d6d/app/src/interfaces/many-to-one/index.ts

> This is something the "regular" o2m interface should do as well! So instead of a custom interface, that's more than welcome as an update to the core o2m interface 🙂

Agreed!
